proto/cheqd/resource/v2/query.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+
+package cheqd.resource.v2;
+
+import "google/api/annotations.proto";
+import "cheqd/resource/v2/resource.proto";
+
+option go_package = "github.com/cheqd/cheqd-node/x/resource/types/v2";
+
+service Query {
+  rpc Resource(QueryGetResourceRequest) returns (QueryGetResourceResponse) {
+    option (google.api.http).get = "/1.0/identifiers/{collection_id}/resources/{id}";
+  }
+  rpc ResourceMetadata(QueryGetResourceMetadataRequest) returns (QueryGetResourceMetadataResponse) {
+    option (google.api.http).get = "/1.0/identifiers/{collection_id}/resources/{id}/metadata";
+  }
+  rpc CollectionResources(QueryGetCollectionResourcesRequest) returns (QueryGetCollectionResourcesResponse) {
+    option (google.api.http).get = "/1.0/identifiers/{collection_id}/resources/all";
+  }
+  rpc AllResourceVersions(QueryGetAllResourceVersionsRequest) returns (QueryGetAllResourceVersionsResponse) {
+    option (google.api.http).get = "/1.0/identifiers/{collection_id}/resources/versions/{name}";
+  }
+}
```

Comment:
So I would simplify this. You don't need to handle all of these as a ledger API, this is the API path that our DID Resolver needs to support. From the ledger, just have a fetch for resource and fetch collection metadata endpoints.

So I'd say:

- `cheqd/resource/v2/resource/<resource-id>` fetch resource
- `cheqd/resource/v2/metadata/<collection-id-OR-resource-id>` fetch collection/resource metadata

What do you think?

Reply:
> cheqd/resource/v2/resource/<resource-id>

This is hard because we have index by `collection id` and only then by `resource id`. There is no global index by `resource id`.

> cheqd/resource/v2/metadata/<collection-id-OR-resource-id>

Same reason here + there is no way to find out which ID is provided - of a collection or of a resource.